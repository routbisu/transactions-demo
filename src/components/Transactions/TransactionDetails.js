import React from 'react';
import styled from 'styled-components';
import Modal from '../common/Modal';
import PropTypes from 'prop-types';
import { Table } from '../../components/common/SimpleTable';
import { formatDate } from '../../services/utils';

const TransactionBody = styled.div`
  padding-top: 20px;
  overflow-x: auto;
`;

export const TransactionDetails = ({ onClose, data }) => {
  const {
    from,
    transaction_date,
    transaction_processed,
    amount,
    description,
  } = data;

  return (
    <Modal onClose={onClose}>
      <TransactionBody>
        <Table className="tablet-and-desktop-only">
          <tbody>
            <tr>
              <th>From</th>
              <td>{from}</td>
            </tr>
            <tr>
              <th>Date</th>
              <td>{formatDate(transaction_date)}</td>
            </tr>
            <tr>
              <th>Amount</th>
              <td>{amount}</td>
            </tr>
            <tr>
              <th>Processed</th>
              <td>{transaction_processed ? '✔️' : '✖️'}</td>
            </tr>
            <tr>
              <td colSpan={2} style={{ whiteSpace: 'normal' }}>
                {description}
              </td>
            </tr>
          </tbody>
        </Table>

        <Table className="mobile-only">
          <tbody>
            <tr>
              <th>From</th>
            </tr>
            <tr>
              <td>{from}</td>
            </tr>
            <tr>
              <th>Date</th>
            </tr>
            <tr>
              <td>{formatDate(transaction_date)}</td>
            </tr>
            <tr>
              <th>Amount</th>
            </tr>
            <tr>
              <td>{amount}</td>
            </tr>
            <tr>
              <th>Processed</th>
            </tr>
            <tr>
              <td>{transaction_processed ? '✔️' : '✖️'}</td>
            </tr>
            <tr>
              <td style={{ whiteSpace: 'normal' }}>{description}</td>
            </tr>
          </tbody>
        </Table>
      </TransactionBody>
    </Modal>
  );
};

TransactionDetails.propTypes = {
  onClose: PropTypes.func,
  data: PropTypes.any,
};

export default TransactionDetails;
