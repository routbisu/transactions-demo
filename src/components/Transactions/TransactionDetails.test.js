import React from 'react';
import { shallow } from 'enzyme';
import { TransactionDetails } from './TransactionDetails';

const tranDataApproved = {
  id: 1,
  account_id: 1,
  description:
    'Dolore quis ad et mollit nisi excepteur ex anim fugiat quis ipsum exercitation proident cupidatat. Quis anim incididunt excepteur cupidatat aliquip nulla reprehenderit velit. Dolor pariatur velit consectetur dolore aliquip reprehenderit non aliqua consectetur. Sunt aliquip consequat et in eu aute.\r\n',
  from: 'Savings Account - 342423455344',
  transaction_date: '2019-06-08T03:37:28 -08:00',
  transaction_processed: true,
  amount: '$1,373.41',
};

const tranDataRejected = {
  id: 1,
  account_id: 1,
  description:
    'Dolore quis ad et mollit nisi excepteur ex anim fugiat quis ipsum exercitation proident cupidatat. Quis anim incididunt excepteur cupidatat aliquip nulla reprehenderit velit. Dolor pariatur velit consectetur dolore aliquip reprehenderit non aliqua consectetur. Sunt aliquip consequat et in eu aute.\r\n',
  from: 'Savings Account - 342423455344',
  transaction_date: '2019-06-08T03:37:28 -08:00',
  transaction_processed: false,
  amount: '$1,373.41',
};

describe('TransactionDetails', () => {
  it('render correctly with transaction approved', () => {
    const component = shallow(<TransactionDetails data={tranDataApproved} />);
    expect(component).toMatchSnapshot();
  });

  it('render correctly with transaction rejected', () => {
    const component = shallow(<TransactionDetails data={tranDataRejected} />);
    expect(component).toMatchSnapshot();
  });
});
