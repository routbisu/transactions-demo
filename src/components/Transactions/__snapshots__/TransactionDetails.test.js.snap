// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetails render correctly with transaction approved 1`] = `ShallowWrapper {}`;

exports[`TransactionDetails render correctly with transaction rejected 1`] = `ShallowWrapper {}`;
